docs/typescript.md

```diff
@@ -0,0 +1,53 @@
+# Typescript
+
+This guide shows you an example of how to develop a function in Typescript with
+the Functions Framework.
+
+1. Use [gts](https://github.com/google/gts) to configure Typescript.
+
+    ```sh
+    npx gts init
+    ```
+
+2. Install the required package(s):
+
+    ```sh
+    npm install @google-cloud/functions-framework
+    ```
+
+    In this example we are using an HTTP signature and require express
+    types:
+
+    ```sh
+    npm install @types/express -D
+    ```
+
+3. Add a `start` script to `package.json`, passing in the `--source` flag to
+   point to the compiled code directory:
+
+    ```js
+      "scripts": {
+        "start": "functions-framework --source=build/src/ --target=helloWorld",
+        ...
+      }
+    ```
+
+4. Create a `src/index.ts` file with the following contents:
+
+    ```ts
+    import type { HttpFunction } from '@google-cloud/functions-framework/build/src/functions';
+
+    export const helloWorld: HttpFunction = (req, res) => {
+      res.send('Hello, World');
+    };
+    ```
+
+5. Compile and start the built-in local development server:
+
+    ```sh
+    $ npm run compile && npm start
+    ...
```

Comment:
- Where do we define the `compile` script?
- We probably want to separate the compiler and function starter. For example, we'd probably want to run `tsc --watch` or something similar in a `watch` script.
- We should not have the `$` symbol in the docs.

Reply:
re: The `compile` script, it is added by `gts`, should that be mentioned?

Reply:
oh, yeah maybe mention it to not confuse people. Something like:

> The `npm run compile` script is defined by `gts`.

Reply:
I'm still developing my first function with this framework and have not gotten around to using `tsc --watch` yet but will add it to the doc. Quick question though, is this used like `nodemon`, i.e. when developing only?

Reply:
Ok just read up and it is actually even defined as the `nodemon` for typescript, nice!